app/main.py

```diff
@@ -1,3 +1,14 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable, Any
+
+
+def cache(func: Callable) -> Callable:
+    results = {}
+
+    def inner(*args) -> Any:
+        if args not in results:
+            print("Calculating new result")
+            results[args] = func(*args)
+            return results[args]
+        else:
+            print("Getting from cache")
```

Comment:
```suggestion
        else:
            print("Getting from cache")
        return results[args]
```
In else case you does not return anything from function